pkg/network/ebpf/c/protocols/http2/decoding-tls.h

```diff
@@ -921,6 +921,12 @@ int uprobe__http2_tls_eos_parser(struct pt_regs *ctx) {
             __sync_fetch_and_add(&http2_tel->end_of_stream, 1);
         }
         handle_end_of_stream(current_stream, &http2_ctx->http2_stream_key, http2_tel);
+
+        // It is not possible that path will be with length 0.
+        if ((current_stream->path.length == 0) && (!current_stream->path.finalized)) {
```

Comment:
I think we should work on the comment here, as it can be misleading reading it for the first time

Something like:

```
// If we reached here, it means that we saw End Of Stream. If the End of Stream came from a request,
// thus we except it to have a valid path. If the End of Stream came from a response, we except it to 
// be after seeing a request, thus it should have a path as well.
```

Regardless, checking `current_stream->path.finalized == false` is sufficient and covers the other condition you added.

also, the same goes for method as well. Reaching up here, means that we should have had both method and a path.
another thing to test here, having `current_stream->response_last_seen != 0` but unfinalized status code

Reply:
I have added the logic for the method and the path. Regarding the logic of the status code versus the last-seen response, I believe it should be in a separate PR and tested to validate its accuracy. I will do so.